compiler/transf.nim

```diff
@@ -321,6 +321,35 @@ proc introduceNewLocalVars(c: PTransf, n: PNode): PTransNode =
     for i in countup(0, sonsLen(n)-1):
       result[i] = introduceNewLocalVars(c, n.sons[i])
 
+proc transformAsgn(c: PTransf, n: PNode): PTransNode =
+  let rhs = n[1]
+
+  if rhs.kind != nkTupleConstr:
+    return transformSons(c, n)
+
+  # Unpack the tuple assignment into N temporary variables and then pack them
+  # into a tuple: this allows us to get the correct results even when the rhs
+  # depends on the value of the lhs
```

Comment:
But we only have to do this when there is aliasing involved, could check for that here or enable my move "optimizer" in the backends (which I intend to do anyway).

Reply:
Fine with me but then I'll merge it once the backend does the heavy lifting, the C backend doesn't understand it well enough to do much (think about ``copyString`` etc).